lib/compress/zstd_compress.c

```diff
@@ -4134,10 +4224,14 @@ size_t ZSTD_compressStream2( ZSTD_CCtx* cctx,
         if (cctx->cdict)
             params.compressionLevel = cctx->cdict->compressionLevel; /* let cdict take priority in terms of compression level */
         DEBUGLOG(4, "ZSTD_compressStream2 : transparent init stage");
-        if (endOp == ZSTD_e_end) cctx->pledgedSrcSizePlusOne = input->size + 1;  /* auto-fix pledgedSrcSize */
+        /* auto-fix pledgedSrcSize when we know we have the entire input buffer available.
+         * NOTE: It is possible to have a stable input buffer without ZSTD_e_end.
+         * E.g. If the user wants to compress with small blocks.
+         */
+        if (endOp == ZSTD_e_end || params.inBufferMode == ZSTD_bm_stable)
```

Comment:
This actually doesn't make sense. I will revert this change. Instead we should require `ZSTD_e_end` to be used if `ZSTD_c_stableInBuffer` is set, or support growing the `in.size` in stable mode.